### PR TITLE
Tabular: Refactored distillation

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Refactored distillation
- Now distillation also creates a weighted ensemble
- Fixed rare bug related to NaN validation scores in Neural Network

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-937-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-937/1/index.html

File: tabular/src/autogluon/tabular/trainer/model_presets/presets.py [hunk 1]
```diff
@@ -299,59 +299,31 @@ def model_factory(
...
+    models, model_args_fit = get_preset_models(problem_type=SOFTCLASS, eval_metric=soft_log_loss,
+                                               hyperparameters=hyperparameters_standard,
+                                               default_priorities=DEFAULT_SOFTCLASS_PRIORITY, invalid_model_names=invalid_model_names, **kwargs)
     if len(models) == 0:
         raise ValueError("At least one of the following model-types must be present in hyperparameters: ['GBM','CAT','NN','RF'], "
                          "These are the only supported models for softclass prediction problems. "
                          "Softclass problems are also not yet supported for fit() with per-stack level hyperparameters.")
     for model in models:
-        model.normalize_pred_probas = True
+        model.normalize_pred_probas = True  # FIXME: Do we need to do this for child models too?
```
Comment (jwmueller) [inline tabular/src/autogluon/tabular/trainer/model_presets/presets.py:327]: I believe so unless they inherit this from parent model

**Review by jwmueller — APPROVED** [review]
looks good! If we're going to automatically construct weighted ensemble in distill(), could it also be worth automatically constructing alternative ensembles in distill() as well via: `fit_weighted_ensemble(base_models = DSTL_models, expand_pareto_frontier=True)`

**Comment by Innixma** [comment]
> looks good! If we're going to automatically construct weighted ensemble in distill(), could it also be worth automatically constructing alternative ensembles in distill() as well via: `fit_weighted_ensemble(base_models = DSTL_models, expand_pareto_frontier=True)`

I had the same idea, but we can leave that for a future PR. It is definitely worth adding as an option.

**Pull request merged**